### PR TITLE
Write up recommended way to test and link it from the readme (Issue 2973)

### PR DESCRIPTION
**Pull request by FrankRay78**
Substantial documentation updates primarily to the BUILDING.md file to provide better guidance to developers wishing to contribute code to NUnit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Documentation for all NUnit projects are hosted on GitHub at [https://github.com
 
 ## Contributing ##
 
-For more information on contributing to the NUnit project, please see [CONTRIBUTING.md](https://github.com/nunit/nunit/blob/master/CONTRIBUTING.md) and the [Developer Docs](https://github.com/nunit/docs/wiki/Team-Practices#technical-practices).
+For more information on contributing to the NUnit project, please see [CONTRIBUTING.md](https://github.com/nunit/nunit/blob/master/CONTRIBUTING.md) and the [Developer Docs](https://docs.nunit.org/articles/developer-info/Team-Practices.html).
```
Comment (mikkelbu) [inline README.md:32]:
```suggestion
For more information on contributing to the NUnit project, please see [CONTRIBUTING.md](https://github.com/nunit/nunit/blob/master/CONTRIBUTING.md) and the [Developer Docs](https://docs.nunit.org/articles/developer-info/Team-Practices.html#technical-practices).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -50,14 +50,13 @@ To help new contributors get their feet wet, we have marked a number of issues w
 
 It is also a good idea to add a comment to an issue that you are working on to let everyone know. If you stop working on it, also please let us know.
 
-Please read through the [developer docs](https://github.com/nunit/docs/wiki/Team-Practices#technical-practices) before contributing to understand our coding standards and contribution guidelines.
+Please read through the [Developer Docs](https://docs.nunit.org/articles/developer-info/Team-Practices.html) before contributing to understand our coding standards and contribution guidelines.
```
Comment (mikkelbu) [inline CONTRIBUTING.md:53]:
```suggestion
Please read through the [Developer Docs](https://docs.nunit.org/articles/developer-info/Team-Practices.html#technical-practices) before contributing to understand our coding standards and contribution guidelines.
```

File: BUILDING.md [hunk 3]
```diff
@@ -1,29 +1,98 @@
...
 
-NUnit 3 consists of three separate layers: the Framework, the Engine and the Console Runner. The source code is kept in two GitHub repositories at https://github.com/nunit/nunit and https://github.com/nunit/nunit-console.
+NUnit 3 consists of three separate layers: the Framework, the Engine and the Console Runner. The source code is kept in two GitHub repositories, https://github.com/nunit/nunit and https://github.com/nunit/nunit-console.
 
-There are two ways to build NUnit: using the solution file in an IDE or through the build script. See also [Building and testing for Linux on a Windows machine](#building-and-testing-for-linux-on-a-windows-machine).
+## Source Code
+
+You will not have permission to create branches directly in either of the above repositories so you will need to work on a local fork. 
+
+The development process for making code changes and submitting Pull Requests (PR) is as follows:
```
Comment (mikkelbu) [inline BUILDING.md:9]:
```suggestion
The development process for making code changes and submitting Pull Requests (PRs) is as follows:
```

File: BUILDING.md [hunk 4]
```diff
@@ -1,29 +1,98 @@
...
+          net45
+          net46
+          netcoreapp2.1
+          netcoreapp3.1
+          netstandard2.0
+```
+
+### Notes
+1. MonoDevelop 2.0 introduced support for multiple target frameworks up to and including .Net 3.0 and 3.5 ([MonoDevelop 2.0 release notes](https://www.monodevelop.com/documentation/release-notes/monodevelop-2.0-released/)). Unfortunately, subsequent releases have not extended that support to .Net Core or .Net 5 and beyond. There is a Kanban board maintained for [MonoDevelop Multi-targeting Support](https://github.com/mono/monodevelop/projects/1) however it's showing the most recent update being 9 Aug 2019. MonoDevelop Version 7.8.4 (build 2) was verified as unable to build `nunit.sln` due to lack of multi-targeting support. This footnote was correct as of 9 June 2021.
+2. GitHub repository cloned and built on 9 June 2021
```
Comment (mikkelbu) [inline BUILDING.md:48]:
```suggestion
2. GitHub repository cloned and built on 9 June 2021.
```

File: BUILDING.md [hunk 5]
```diff
@@ -1,29 +1,98 @@
...
+### Workarounds
 
-We use [Cake](https://cakebuild.net) to build NUnit for distribution. The primary script that controls building, running tests and packaging is `build.cake`. We modify `build.cake` when we need to add new targets or change the way the build is done. Normally `build.cake` is not invoked directly but through `build.ps1` (on Windows) or `build.sh` (on Linux). These two scripts are provided by the Cake project and ensure that Cake is properly installed before trying to run the cake script.
+#### 1. NUnit Lite Runner (run from command prompt / terminal)
+
+The NUnit solution contains [NUnit Lite Runner](https://docs.nunit.org/articles/nunit/running-tests/NUnitLite-Runner.html), a lightweight test runner. It is similar to NUnit console but with fewer features and without the overhead of a full NUnit installation. 
+
+If you navigate to one of the build outputs under the `bin` directory, you will see that it contains `nunitlite-runner.exe`
+
+For example, the following command `./nunitlite-runner nunit.framework.tests.dll` (bash on Linux) will run all tests in the nunit.framework.tests.dll
```
Comment (mikkelbu) [inline BUILDING.md:75]:
```suggestion
For example, the following command `./nunitlite-runner nunit.framework.tests.dll` (bash on Linux) will run all tests in the nunit.framework.tests.dll.
```

File: BUILDING.md [hunk 6]
```diff
@@ -1,29 +1,98 @@
...
+
+NUnit Lite Runner accepts a number of [command line arguments](https://docs.nunit.org/articles/nunit/running-tests/NUnitLite-Options.html). 
+
+For example, the following command `./nunitlite-runner nunit.framework.tests.dll --where "class == NUnit.Framework.Assertions.AssertEqualsTests"` (bash on Linux) will run all tests in the NUnit.Framework.Assertions.AssertEqualsTests TextFixture in the nunit.framework.tests.dll
```
Comment (mikkelbu) [inline BUILDING.md:79]:
```suggestion
For example, the following command `./nunitlite-runner nunit.framework.tests.dll --where "class == NUnit.Framework.Assertions.AssertEqualsTests"` (bash on Linux) will run all tests in the NUnit.Framework.Assertions.AssertEqualsTests TextFixture in the nunit.framework.tests.dll.
```

File: BUILDING.md [hunk 7]
```diff
@@ -1,29 +1,98 @@
...
+
+#### 2. NUnit Lite Runner (run by the IDE after a build)
+
+You can run NUnit Lite in a similar way to above, but from the development IDE after a successful build. 
+
+This is probably the easiest way (whilst issue [#3008](https://github.com/nunit/nunit/issues/3008) remains open) for developers to step through code and debug any new contributions made to the codebase.
+
+In your development IDE of choice, ensure there is a Run/Debug configuration which specifies `nunitlite-runner` as the startup project. 
+
+Then specify the program arguments in exactly the same way as above. For example the following argument `nunit.framework.tests.dll --where "class == NUnit.Framework.Assertions.AssertEqualsTests"` will run all tests in the NUnit.Framework.Assertions.AssertEqualsTests TextFixture in the nunit.framework.tests.dll
```
Comment (mikkelbu) [inline BUILDING.md:89]:
```suggestion
Then specify the program arguments in exactly the same way as above. For example the following argument `nunit.framework.tests.dll --where "class == NUnit.Framework.Assertions.AssertEqualsTests"` will run all tests in the NUnit.Framework.Assertions.AssertEqualsTests TextFixture in the nunit.framework.tests.dll.
```

**Review by mikkelbu — APPROVED** [review]
Thanks for providing this PR @FrankRay78. I only have some very minor nit-pick, but overall it looks good 👍. Do you have any preference regarding squash when we merge this?

**Comment by FrankRay78** [comment]
Hi @mikkelbu, I have no preference when you merge this.  Given that I haven't actually contributed any code changes yet, I wanted to draft the above as an improvement to what I struggled with just getting ready to contribute. Happy for you to make whatever further improvements / changes etc - or anything major let me know and I can do a second draft. 

re: the unit test runners in the IDE's, as drafted above I think is an accurate summary of where it's at. I did start to look into getting it working again in VS last night, but it's not straightforward to work out what is going on (for me at least). But I do suspect it's todo with .Net Core 2.1

**Comment by mikkelbu** [comment]
Thanks for providing this improvement to the documentation @FrankRay78 👍.

**Pull request merged**